dot/parachain/dispute/overseer/message.go

```diff
@@ -0,0 +1,83 @@
+package overseer
+
+import (
+	parachainTypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+type ChainAPIMessage struct {
+	RelayParent     common.Hash
+	ResponseChannel chan *uint32
+}
+
+type PersistedValidationData struct {
+	ParentHead             []byte
+	RelayParentNumber      uint32
+	RelayParentStorageRoot common.Hash
+	MaxPOVSize             uint32
+}
+
+type AvailableData struct {
+	POV            []byte
+	ValidationData PersistedValidationData
+}
+
+type RecoveryError uint32
+
+const (
+	RecoveryErrorInvalid RecoveryError = iota
+	RecoveryErrorUnavailable
+)
+
+func (e RecoveryError) String() string {
+	switch e {
+	case RecoveryErrorInvalid:
+		return "invalid"
+	case RecoveryErrorUnavailable:
+		return "unavailable"
+	default:
+		return "unknown"
+	}
+}
+
+type AvailabilityRecoveryResponse struct {
+	AvailableData *AvailableData
+	Error         *RecoveryError
+}
+
+type AvailabilityRecoveryMessage struct {
+	CandidateReceipt parachainTypes.CandidateReceipt
+	SessionIndex     parachainTypes.SessionIndex
+	GroupIndex       *uint32
+	ResponseChannel  chan AvailabilityRecoveryResponse
+}
+
+type PvfExecTimeoutKind uint32
+
+const (
+	PvfExecTimeoutKindBacking PvfExecTimeoutKind = iota
+	PvfExecTimeoutKindApproval
+)
+
+type ValidateFromChainState struct {
+	CandidateReceipt   parachainTypes.CandidateReceipt
+	PoV                []byte
```

Comment:
we have [PoV](https://github.com/ChainSafe/gossamer/blob/f9b3281b3e1798510948499b6a42900334a97edc/dot/parachain/available_data_fetching.go#L62) type. Can you use it here?

Reply:
The types in the overseer is temporary. Please see the comment on top of the file.